### PR TITLE
Spring Shell should also add spring-shell-starter-test

### LINKED ISSUE #1454: Spring Shell should also add spring-shell-starter-test
Selecting Spring Shell should automatically add spring-shell-starter-test as a test dependency.

NOTE: I read through [this](https://github.com/spring-io/initializr/issues/461) and it feels similar.

### PR DESCRIPTION
**Pull request by dashaun**

### CODE REVIEW

File: start-site/src/main/java/io/spring/start/site/extension/dependency/springshell/package-info.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Extensions for generation of projects that depend on Spring Security.
```
Comment (wilkinsona) [inline start-site/src/main/java/io/spring/start/site/extension/dependency/springshell/package-info.java:18]: `Security` -> `Shell`

File: start-site/src/main/java/io/spring/start/site/extension/dependency/springshell/SpringShellTestBuildCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * A {@link BuildCustomizer} that automatically adds {@code spring-shell-test} when Spring
+ * Shell is selected.
+ *
+ * @author DaShaun Carter
+ */
+public class SpringShellTestBuildCustomizer implements BuildCustomizer<Build> {
+
+	@Override
+	public void customize(Build build) {
+		var springShell = Dependency.withCoordinates("org.springframework.shell", "spring-shell-starter-test")
```
Comment (wilkinsona) [inline start-site/src/main/java/io/spring/start/site/extension/dependency/springshell/SpringShellTestBuildCustomizer.java:34]: We don't use `var` in this codebase.

A single statement, as in `SpringSecurityTestBuildCustomizer`, could be used here.

File: start-site/src/test/java/io/spring/start/site/extension/dependency/springshell/SpringShellTestBuildCustomizerTests.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import io.spring.initializr.metadata.BillOfMaterials;
+import io.spring.initializr.metadata.Dependency;
+import io.spring.initializr.web.project.ProjectRequest;
+import io.spring.start.site.extension.AbstractExtensionTests;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SpringShellTestBuildCustomizerTests extends AbstractExtensionTests {
```
Comment (wilkinsona) [inline start-site/src/test/java/io/spring/start/site/extension/dependency/springshell/SpringShellTestBuildCustomizerTests.java:27]: Please add some class-level javadoc similar to [this](https://github.com/spring-io/start.spring.io/blob/3c71d57eae02e099b586a0ee38fc572e5a2f571c/start-site/src/test/java/io/spring/start/site/extension/dependency/springsecurity/SpringSecurityTestBuildCustomizerTests.java#L26-L30).

**Review by wilkinsona — CHANGES_REQUESTED** [review]

File: start-site/src/test/java/io/spring/start/site/extension/dependency/springshell/SpringShellTestBuildCustomizerTests.java (hunk 3 above)
Comment (dashaun) [reply start-site/src/test/java/io/spring/start/site/extension/dependency/springshell/SpringShellTestBuildCustomizerTests.java:32]: @wilkinsona Thanks for holding my hand with this PR.  It has been a pleasure, getting a code review from you!

😄

**Comment by mhalbritter** [comment]
Thanks DaShaun! Keep 'em coming! :)

**Pull request closed without merging**